sql-plugin/src/main/scala/com/nvidia/spark/rapids/DateUtils.scala

```diff
@@ -40,15 +40,15 @@ object DateUtils {
     "u", "uu", "uuu", "uuuu", "uuuuu", "uuuuuu", "uuuuuuu", "uuuuuuuu", "uuuuuuuuu", "uuuuuuuuuu",
     "y", "yyy", "yyyyy", "yyyyyy", "yyyyyyy", "yyyyyyyy", "yyyyyyyyy", "yyyyyyyyyy",
     "D", "DD", "DDD", "s", "m", "H", "h", "M", "MMM", "MMMM", "MMMMM", "L", "LLL", "LLLL", "LLLLL",
-    "d", "S", "SS", "SSS", "SSSS", "SSSSS", "SSSSSSSSS", "SSSSSSS", "SSSSSSSS")
```

Comment:
This change could potentially affect many areas of functionality, such as:

- GpuTextBasedPartitionReader#castStringToDate (used when reading dates from csv/json)
- ToUnixTimestamp / FromUnixTime / UnixTimestamp / GetTimestamp

We'll need to be careful to review tests for these areas to see if we need to test for these newly supported formats to make sure we are compatible with CPU

Reply:
Yeah I agree that this may have more affect. I just realized that the format function alone is converting the input column into a column of `TIMESTAMP_SECONDS` type, which effectively truncates all the subsecond information. So in order to support subsecond format, we have to check and fix in many places too.

Reply:
@andygrove Finally I've completed scanning all these places. 
 * `GpuTextBasedPartitionReader` seems already handles `SSS` format
 * Unix time only stores time down to second thus subsecond information is not important

My changes in this PR also passed all integration tests, thus I don't think there will be any other significant impact.

Reply:
I reviewed `GpuTextBasedPartitionReader` and I agree - it has its own handling already.

I added this test to `TimeOperatorsSuite` and it passed, so that gives me confidence. We don't seem to have any integration tests for `from_unixtime` so we should probably add some (as a follow on).  

```
  testSparkResultsAreEqual(
    "Test from_unixtime with alternative month and two digit year SSS", datesPostEpochDf,
    conf = new SparkConf().set(RapidsConf.INCOMPATIBLE_DATE_FORMATS.key, "true")) {
    frame => frame.select(from_unixtime(col("dates"), "dd/LL/yy HH:mm:ss.SSS"))
  }
```